### PR TITLE
Let's Split Personal keymap added

### PR DESCRIPTION
**Pull request by pnrnc485**

### CODE REVIEW

File: keyboards/lets_split/keymaps/suzuken/config.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/lets_split/keymaps/suzuken/config.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/lets_split/keymaps/suzuken/keymap.c [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+};
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+float tone_dvorak[][2]     = SONG(DVORAK_SOUND);
+float tone_colemak[][2]    = SONG(COLEMAK_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/keymap.c]: Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/lets_split/keymaps/suzuken/keymap.c [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  _______, RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_DEL, \
+  _______, _______, _______, AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______ \
+)
+
+
+};
+
+#ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/keymap.c]: This isn't necessary if using the `set_single_persistent_default_layer` function.

File: keyboards/lets_split/keymaps/suzuken/keymap.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/keymap.c]: This entire section can be replaced with just `set_single_persistent_default_layer(_QWERTY)`

File: keyboards/lets_split/keymaps/suzuken/keymap.c (hunk 4 above)
Comment (drashna) [reply keyboards/lets_split/keymaps/suzuken/keymap.c]: See the documentation to see how to keep the audio feedback: 
https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

eg, add this to your config.h: 
```
#define DEFAULT_LAYER_SONGS { SONG(QWERTY_SOUND), \
                              SONG(COLEMAK_SOUND), \
                              SONG(DVORAK_SOUND) \
                            }
```

File: keyboards/lets_split/keymaps/suzuken/rules.mk [hunk 6]
```diff
@@ -0,0 +1,3 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/lets_split/keymaps/suzuken/rules.mk]: This file can be removed.

Specifically, the contents can be removed completely, as the ifdef block is a holdover from the old make system, and that leaves an empty file.

File: keyboards/lets_split/keymaps/suzuken/keymap.c [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  DVORAK,
+  LOWER,
+  RAISE,
+  ADJUST,
+};
+
+// Fillers to make layering more clear
```
Comment (noroadsleft) [inline keyboards/lets_split/keymaps/suzuken/keymap.c]: Lines 25-27 can be safely deleted as QMK defines these automatically with every keymap.

https://github.com/qmk/qmk_firmware/blob/27e216fc1a481bb3e65832458bd2b199cb499e0b/quantum/quantum_keycodes.h#L29-L31

**Comment by pnrnc485** [comment]
@noroadsleft @drashna 

I am changing the keymap by copying the default, but is it necessary to fix it?

Please correct the defaults if necessary.

**Comment by drashna** [comment]
@pnrnc485 Yes and no.

It's not strictly necessary, but we want to keep new additions up to date with coding conventions, and the like.

We have been going through and updating the boards, but haven't gotten to all of them yet. 

So if you wouldn't mind, please do make the changes.

**Comment by drashna** [comment]
Any update on this?

**Comment by drashna** [comment]
Any update on this?  
Also, it is fine if you don't want to mak ethe changes, we just want to make sure it's up to date with current coding conventions.  But that isn't necessary.

Let us know either way.  Otherwise, we may just close the PR, since it's been open and inactive for a long time, now.

**Comment by drashna** [comment]
There haven't been any updates to this PR in almost a year, despite a couple of pings. 

I'm closing this, for now.  If you wish to update this or get it merged, you can reopen this PR or open a new one.  

Sorry

**Pull request closed without merging**